internal/sys/fs.go

```diff
@@ -65,15 +65,26 @@ var emptyFSContext = &FSContext{
 	lastFD:      2,
 }
 
-// NewFSContext returns a mutable context if the fs is not EmptyFS.
+// NewFSContext returns a mutable context if the fs is not EmptyFS. The file
+// descriptor used in the case of non-empty is 3.
+//
+// Why file descriptor 3?
+//
+// While not specified, the most common WASI implementation, wasi-libc, expects
+// POSIX style file descriptor allocation, where the lowest available number is
+// used to open the next file. Since 1 and 2 are taken by stdout and stderr
+// respectively, this means the `fs` parameter will be 3.
```

Comment:
```suggestion
// respectively, this means the `fd` parameter will be 3 in the first call to `fd_prestat_get`.
```

Mentioning fd_prestat_get here might be TMI, fs I guess is a typo though